### PR TITLE
fix(sqlmap): add input validation to session cookie field (#1759)

### LINKED ISSUE #1759: [Improvement Request]: Add input validation to the SQLMap Session Cookie field
### Short name for suggested Improvement

Input validation for the SQLMap Session Cookie field

### Priority

Low

### Categories

UI, UX, Enhancement

### **Description of Feature**

Follow-up to #1614. The Session Cookie field added in that fix works correctly for well-formed values, but currently performs little or no validation on the string entered. Values are passed directly into the HTTP Cookie: header without a warning when the format is malformed.

During retesting, malformed or questionable values were accepted and forwarded to sqlmap without any DDT-level warning. For example:

test — does not contain the required name=value structure.

login=test/test'; — contains a trailing semicolon without another cookie pair following it.

The tests did not produce a crash or Python traceback, and command construction remained stable. However, because the field already provides a format example such as:

PHPSESSID=abc123; security=low

it would improve usability if clearly malformed cookie input were detected before starting the scan.

### **Expected Behavior**

The Session Cookie field should perform lightweight validation before submitting the scan.

Clearly malformed input, such as a value without a name=value pair, should trigger an inline warning or prevent the scan from starting.

Valid cookie strings containing one or more name=value pairs separated by semicolons should continue to be accepted.

The field should also continue to allow an entirely empty value because the Session Cookie option is optional, as verified during the #1614 retest.

### **Steps to Reproduce**

Open the SQLMap component in DDT.
Enter a valid target URL.
In the Session Cookie field enter test.
Click Start SQLMap.

### **Actual Behaviour**

The malformed value is accepted without a warning.

The scan proceeds and the value is forwarded in the request.

No DDT-level validation error is displayed to the user.

### **Related**

Follow-up improvement identified during retesting of #1614.

### PR DESCRIPTION
**Pull request by Ar1es-XD**
## Description
Resolves #1759.

Follow-up to #1614. The Session Cookie field added in that fix works correctly for well-formed values, but previously lacked input validation, allowing clearly malformed strings (e.g., `test` lacking a `name=value` structure, or `login=test/test';` containing a trailing semicolon) to be accepted and passed directly to SQLMap without any DDT-level warning or prevention.

### Summary of Changes
- **Validation Logic**: Added a `validateSessionCookie` helper function to validate cookie input:
  - Accepts empty or whitespace-only values since the Session Cookie field is optional.
  - Verifies that input consists of valid `name=value` cookie pairs separated by semicolons (e.g., `PHPSESSID=abc123; security=low`).
  - Rejects inputs without `name=value` pairs (e.g., `test`), missing names or values, empty pairs, and leading or trailing semicolons (e.g., `login=test/test';`).
  - Correctly permits cookie values containing special characters, slashes, or base64 padding (e.g., `token=abc==; id=123`).
- **Form Integration**: Wired `validateSessionCookie` into Mantine's `useForm` validation for the `sessionCookie` field:
  - Displays an inline warning under the Session Cookie field if input is malformed.
  - Halts submission in `onSubmit` if validation errors are present, preventing the scan from starting with invalid cookies.
- **Argument Trimming**: Ensured `values.sessionCookie.trim()` is passed to `--cookie` so accidental outer whitespace is stripped.

## Testing Evidence
- Locally verified the validation logic across all edge cases including valid formats (empty input, single pair, multiple pairs, base64 values, special characters) and invalid formats (missing name=value pairs, trailing semicolon, leading semicolon, empty pairs, spaces in names).
- Verified `yarn run style` passes with Prettier formatting.
- Verified production build succeeds (`yarn run build`).
- Verified existing test suite passes (`yarn test`).

Closes #1759

### CODE REVIEW

File: src/components/SQLmap/SQLmap.tsx [hunk 1]
```diff
@@ -26,6 +26,53 @@ interface FormValuesType {
...
 
+/**
+ * Validates the session cookie string for SQLMap.
+ * Accepts an empty string (optional field) or valid cookie format:
+ * one or more name=value pairs separated by semicolons.
+ *
+ * @param {string} cookie - The session cookie string to validate.
+ * @returns {string | null} Error message if invalid, null if valid.
+ */
+export const validateSessionCookie = (cookie: string): string | null => {
```
[the remark this answers is not in the document]
Comment (Ar1es-XD) [reply src/components/SQLmap/SQLmap.tsx:37]: Updated the PR description to clarify that testing was performed locally across all edge cases rather than committed as a separate test file, following repository conventions.

**Review by typicalskip — APPROVED** [review]
Looks good to me
